analysis/feasibility.R

```diff
@@ -0,0 +1,121 @@
+       y = "Number of Cases") +
+  scale_color_manual(values = c("t2dm" = "blue", "t1dm" = "red", "gestationaldm" = "green", "pcos" = "yellow", "predm" = "orange")) + 
+  theme_minimal()
+
+## 2. How many with a T2DM diagnosis start metformin and median time to start ?
+# T2DM variable from diabetes algo
+n_t2dm_metfin_start <- data_processed %>%
+  filter(!is.na(cov_date_t2dm)) %>%
+  filter(!is.na(exp_date_metfin_first)) %>%
+  filter(exp_date_metfin_first > cov_date_t2dm) %>%
+  mutate(tb_T2DMdiag_metfin = as.numeric(difftime(exp_date_metfin_first, cov_date_t2dm, units = "days"))) %>%
+  summarise(
+    n_start_metfin_aT2DM = n(),
+    median_tb_T2DMdiag_metfin = median(tb_T2DMdiag_metfin, na.rm = TRUE),
+    IQR_lower = quantile(tb_T2DMdiag_metfin, 0.25, na.rm = TRUE),  
+    IQR_upper = quantile(tb_T2DMdiag_metfin, 0.75, na.rm = TRUE))
+# gt table
+tbl_n_t2dm_metfin_start <- n_t2dm_metfin_start %>%
+  gt() %>%
+  tab_header(title = "Metformin start after T2DM diagnosis") %>%
+  cols_label(
+    n_start_metfin_aT2DM = "N starting metformin after T2DM diagnosis",
+    median_tb_T2DMdiag_metfin = "Median time to metformin start (days)",
+    IQR_lower = "IQR Lower",
+    IQR_upper = "IQR Upper")
+
+## 2. How many with a T2DM diagnosis start metformin within 10d after (the first) pos. SARS-CoV-2 test
+# T2DM variable from diabetes algo
+n_t2dm_covid_metfin_start <- data_processed %>%
+  filter(!is.na(cov_date_t2dm)) %>%
+  filter(!is.na(exp_date_metfin_first)) %>%
+  filter(exp_date_metfin_first > cov_date_t2dm) %>%
+  filter(!is.na(cov_date_covid19_first)) %>%
+  filter(exp_date_metfin_first <= (cov_date_covid19_first + days(10))) %>% # grace period 10 days
+  mutate(tb_T2DMdiag_metfin = as.numeric(difftime(exp_date_metfin_first, cov_date_t2dm, units = "days"))) %>%
+  mutate(tb_COVIDdiag_metfin = as.numeric(difftime(exp_date_metfin_first, cov_date_covid19_first, units = "days"))) %>%
+  summarise(
+    n_start_metfin_aCOVID = n(),
+    median_tb_T2DMdiag_metfin = median(tb_T2DMdiag_metfin, na.rm = TRUE),
+    IQR_lower = quantile(tb_T2DMdiag_metfin, 0.25, na.rm = TRUE),  
+    IQR_upper = quantile(tb_T2DMdiag_metfin, 0.75, na.rm = TRUE))
+# gt table
+tbl_n_t2dm_covid_metfin_start <- n_t2dm_covid_metfin_start %>%
+  gt() %>%
+  tab_header(title = "Metformin start within 10d after COVID diagnosis among T2DM") %>%
+  cols_label(
+    n_start_metfin_aCOVID = "N starting metformin 10d after COVID diagnosis",
+    median_tb_T2DMdiag_metfin = "Median time to metformin start after T2DM",
+    IQR_lower = "IQR Lower",
+    IQR_upper = "IQR Upper")
+
+################################################################################
+# 3 Save output
+################################################################################
+ggsave(plot_incidence, filename = here::here("output", "data_properties", "plot_incidence.png"))
+gtsave(tbl_n_t2dm_metfin_start, filename = here::here("output", "data_properties", "tbl_n_t2dm_metfin_start.html"))
```

Comment:
You need to apply disclosure control to your output - [documentation here](https://docs.opensafely.org/outputs/sdc/). We usually use midpoint6 rounding.

Reply:
thanks! Should be fine now.